src/progress/progress_set.rs

```diff
@@ -159,16 +161,22 @@ pub struct ProgressSet {
     // You should not depend on these values unless you just set them.
     // We use a cell to avoid taking a `&mut self`.
     sort_buffer: RefCell<Vec<u64>>,
+    logger: Logger,
 }
 
 impl ProgressSet {
     /// Creates a new ProgressSet.
-    pub fn new() -> Self {
-        Self::with_capacity(0, 0)
+    pub fn new<'a>(logger: impl Into<Option<&'a Logger>>) -> Self {
```

Comment:
How about `fn logger<'a>(&mut self, logger: impl Into<Option<&'a Logger>>) -> Self {...}`? This can help preserve API and avoid the `None` stuff we now see in some tests.